satellite/admin/user.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (C) 2020 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package admin
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"golang.org/x/crypto/bcrypt"
+
+	"storj.io/common/uuid"
+	"storj.io/storj/satellite/console"
+)
+
+func (server *Server) addUser(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to read body: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	var input console.CreateUser
+
+	err = json.Unmarshal(body, &input)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to unmarshal request: %v", err), http.StatusBadRequest)
```

Comment:
Maybe this is a internal error?

Reply:
Since this is from the end-user which should match the struct, the only error can be in the request.